autoextract_poet/page_inputs.py

```diff
@@ -0,0 +1,59 @@
+from typing import ClassVar, List, Optional, Type
+
+import attr
+
+from autoextract_poet.items import (
+    Article,
+    Item,
+    Product,
+)
+
+
+@attr.s(auto_attribs=True)
+class ResponseData:
+    """Container for AutoExtract response data.
+
+    Should not be used directly by providers.
+    Use derived classes like ArticleResponseData and similar.
+
+    API responses are wrapped in a JSON array
+    (this is to facilitate query batching).
+
+    https://doc.scrapinghub.com/autoextract.html#responses
+    """
+
+    item_class: ClassVar[Type[Item]]
+    item_key: ClassVar[str]
+
+    data: dict
+
+    def to_items(self) -> Optional[List[Item]]:
+        return self.item_class.from_list(
+            [
+                query_result[self.item_key]
+                for query_result in self.data
+                if "error" not in query_result
```

Comment:
I don't think errors should be handled here. It should be the provider how checks this in advance and if error, then raise an Exception, so this method would be never invoked. Also as this should only return a single item, it won't be a problem having a mixture of both items and errors.

Reply:
This piece of code was removed.